java/src/org/exist/jms/replication/subscribe/ReplicationJmsListener.java

```diff
@@ -729,9 +720,9 @@ private Collection getOrCreateCollection(final XmldbURI sourcePath, final Option
             throw new MessageReceiveException(t.getMessage(), t);
 
         } finally {
-            releaseLock(newCollection, Lock.LockMode.WRITE_LOCK);
+            releaseLock(collection, Lock.LockMode.WRITE_LOCK);
```

Comment:
Yeah the documentation sucks, you really have to dig deep into the code to understand. I am currently replacing the locking implementation in eXist with something more consistent.

What you need to know most likely right now is that:
1. `NativeBroker#openCollection` returns a locked Collection, you must manually release that lock when you are finished with the Collection.
2. `NativeBroker#getCollection` returns an unlocked Collection.
3. `NativeBroker#getOrCreateCollection` returns a unlocked Collection or locked Collection(s) depending on whether get or create occurs; creation will write lock all the collections in the path that it needs to create. You DO NOT need to manually release these locks as ownership is passed to the Transaction which will auto-free them.